core/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -137,6 +137,23 @@ public T value()
     return value;
   }
 
+  void setStringValue(@Nullable String value)
+  {
+    stringValue = value;
+    stringValueValid = true;
+  }
+
+  @Nullable
+  String getStringValue()
+  {
+    return stringValue;
+  }
+
+  boolean isStringValueValid()
+  {
+    return stringValueValid;
```

Comment:
This method is to check whether or not `stringValue` is initialized. It will explode [this line](https://github.com/apache/druid/pull/9812/files/abe6a6bb03fa21ee07819c01d9edb62180c06d10#diff-03e48730fe9116ce0c8c99610d4e37f0R591).

Reply:
Sorry, I meant to add this comment to `getStringValue()`.

Reply:
Ah good point. Added an assert. Also renamed `getStringValue()` and `isStringValueValid()` to `getStringValueCached()` and `isStringValueCached()`.